File: desktop/sidecar/daemon/schemas/review.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel

from .git import GitDiffResult


ReviewFileStatus = Literal["added", "modified", "deleted", "renamed"]
CommitMessageStatus = Literal["generated", "unavailable", "failed"]


class ReviewFile(BaseModel):
    path: str
    old_path: str | None = None
    status: ReviewFileStatus
    staged: bool
    unstaged: bool
    untracked: bool
    insertions: int
    deletions: int


class ReviewSummary(BaseModel):
    files_changed: int
    insertions: int
    deletions: int
    staged_count: int
    unstaged_count: int
    untracked_count: int


class ReviewFilesResult(BaseModel):
    files: list[ReviewFile]
    summary: ReviewSummary
    working_dir: str
    branch: str


class ReviewDiffRequest(BaseModel):
    path: str | None = None
    staged: bool = False


class ReviewPathsRequest(BaseModel):
    paths: list[str]


class ReviewCommitRequest(BaseModel):
    message: str


class ReviewCommitMessageRequest(BaseModel):
    avoid: str | None = None


class ReviewCommitMessageResult(BaseModel):
    status: CommitMessageStatus
    message: str | None = None
    detail: str | None = None


class ReviewOkResult(BaseModel):
    ok: bool
    detail: str | None = None


class ReviewPrResult(BaseModel):
    ok: bool
    url: str | None = None
    detail: str | None = None


class ReviewDiffResult(GitDiffResult):
    pass

File: apps/hermes-studio/src/types/review.ts
import type { FileStatus, GitDiffResult } from './diff.js';

export interface ReviewFile {
  path: string;
  old_path: string | null;
  status: FileStatus;
  staged: boolean;
  unstaged: boolean;
  untracked: boolean;
  insertions: number;
  deletions: number;
}

export interface ReviewSummary {
  files_changed: number;
  insertions: number;
  deletions: number;
  staged_count: number;
  unstaged_count: number;
  untracked_count: number;
}

export interface ReviewFilesResult {
  files: ReviewFile[];
  summary: ReviewSummary;
  working_dir: string;
  branch: string;
}

export interface ReviewCommitMessageResult {
  status: 'generated' | 'unavailable' | 'failed';
  message: string | null;
  detail?: string | null;
}

export interface ReviewShipInfoResult {
  current_branch: string;
  default_branch: string | null;
  pr_url: string | null;
  gh_available: boolean;
  can_create_pr: boolean;
}

export interface ReviewOkResult {
  ok: boolean;
  detail?: string | null;
}

export interface ReviewPrResult {
  ok: boolean;
  url?: string | null;
  detail?: string | null;
}

export type ReviewDiffResult = GitDiffResult;
